checks.d/iis.py

```diff
@@ -1,114 +1,154 @@
+            for wmi_property, wmi_value in wmi_obj.iteritems():
+                # Tag with `tag_by` parameter
+                try:
+                    metrics.append(WMIMetric(wmi_property, float(wmi_value), tags))
+                except ValueError:
+                    self.log.warning(u"When extracting metrics with WMI, found a non digit value"
+                                     " for property '{0}'.".format(wmi_property))
+                    continue
+                except TypeError:
+                    self.log.warning(u"When extracting metrics with WMI, found a missing property"
+                                     " '{0}'".format(wmi_property))
+                    continue
+        return metrics
+
+    def _submit_events(self, wmi_sampler, sites):
+        expected_sites = set(sites)
+
+        for wmi_obj in wmi_sampler:
+            sitename = wmi_obj['Name']
+            if sitename == "_Total":
                 continue
 
-            # Tag with the site name if we're not using the aggregate
-            if iis_site.Name != '_Total':
-                tags = instance_tags + ['site:%s' % iis_site.Name]
-            else:
-                tags = instance_tags
-
-            status = AgentCheck.CRITICAL if iis_site.ServiceUptime == 0 else AgentCheck.OK
-            self.service_check("iis.site_up", status, tags=['site:%s' % iis_site.Name])
-            expected_sites.remove(iis_site.Name)
-
-            for metric, mtype, wmi_val in self.METRICS:
-                if not hasattr(iis_site, wmi_val):
-                    if wmi_val == 'TotalBytesTransferred' and hasattr(iis_site,
-                                                                      'TotalBytesTransfered'):
-                        # Windows 2008 sp2 reports it as TotalbytesTransfered
-                        # instead of TotalBytesTransferred (single r)
-                        wmi_val = 'TotalBytesTransfered'
-                    elif wmi_val == 'TotalConnectionAttemptsAllInstances' \
-                            and hasattr(iis_site, 'TotalConnectionAttemptsallinstances'):
-                        wmi_val = 'TotalConnectionAttemptsallinstances'
-                    else:
-                        self.warning("Unable to fetch metric %s. Missing %s in "
-                                     "Win32_PerfFormattedData_W3SVC_WebService"
-                                     % (metric, wmi_val))
-                        continue
-
-                # Submit the metric value with the correct type
-                value = float(getattr(iis_site, wmi_val))
-                metric_func = getattr(self, mtype)
-                metric_func(metric, value, tags=tags)
-
-        for remaining_site in expected_sites:
+            uptime = wmi_obj["ServiceUptime"]
+            status = AgentCheck.CRITICAL if uptime == 0 else AgentCheck.OK
+
+            self.service_check("iis.site_up", status, tags=['site:{0}'.format(self.normalize(sitename))])
```

Comment:
I added c35d693 to deal with it the case sitename was unicode encoded, because normalize wasn't ready for that and I wasn't sure what ExecQuery returned. I felt it probably made sense to normalize to ascii, it's probably what one would expect when calling normalize on a unicode string.